tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11138,25 +11138,44 @@ bool Sema::DiagnoseHLSLMethodCall(const CXXMethodDecl *MD, SourceLocation Loc) {
         Diags.Report(Loc, diag::err_hlsl_wg_nodetrackrwinputsharing_missing);
         return true;
       }
-    } else if (opCode == hlsl::IntrinsicOp::MOP_CalculateLevelOfDetail ||
-               opCode ==
-                   hlsl::IntrinsicOp::MOP_CalculateLevelOfDetailUnclamped) {
-      const auto *shaderModel =
-          hlsl::ShaderModel::GetByName(getLangOpts().HLSLProfile.c_str());
-      if (!shaderModel->IsSM68Plus()) {
+    }
+  }
+  return false;
+}
+
+// Check HLSL member call constraints for used functions.
+bool Sema::DiagnoseUsedHLSLMethodCall(const CXXMethodDecl *MD,
+                                      SourceLocation Loc,
+                                      const hlsl::ShaderModel *SM,
+                                      DXIL::ShaderKind EntrySK) {
+  bool bError = false;
+  if (MD->hasAttr<HLSLIntrinsicAttr>()) {
+    hlsl::IntrinsicOp opCode =
+        (IntrinsicOp)MD->getAttr<HLSLIntrinsicAttr>()->getOpcode();
+
+    if (opCode == hlsl::IntrinsicOp::MOP_CalculateLevelOfDetail ||
```

Comment:
Should we make this a switch statement instead for future additions to this function?

Reply:
Updated.